components/datetime/src/fields/macros.rs

```diff
@@ -83,14 +95,19 @@ macro_rules! field_type {
             /// This is mostly useful for serialization,
             /// and does not guarantee index stability between ICU4X
             /// versions.
-            pub(crate) fn _from_idx(idx: u8) -> Result<Self, SymbolError> {
+            pub(crate) fn from_idx(idx: u8) -> Result<Self, SymbolError> {
                 match idx {
                     $(
                         $idx => Ok(Self::$val),
                     )*
                     _ => Err(SymbolError::InvalidIndex(idx)),
                 }
             }
+
+            pub(crate) fn idx_in_range(v: &u8) -> bool {
```

Comment:
issue: this doesn't seem to be accurate if the enum is written without sequential discriminant values

Reply:
Perhaps we should try and forbid specifying the enum with anything other than sequential discriminant values? Or autogenerate them. Both of these are tricky to do as macros.

Reply:
@iainireland does this using a new dependency on [num_enum](https://docs.rs/num_enum/0.5.4/num_enum/) in  #1089.

Reply:
Yeah perhaps we should do the same thing here: as it stands it's a bit unsafe.

Reply:
I followed Manish short-term advise and renamed to `unsafe_field_type` and I agree with Shane that we should solve it in ULE context, not per customer. Filed #1123 to track it.

Reply:
Filed https://github.com/unicode-org/icu4x/issues/1123 and renamed here to `unsafe_field_type`.